packages/@aws-cdk/aws-servicecatalogappregistry/lib/application-associator.ts

```diff
@@ -17,58 +17,86 @@ export interface ApplicationAssociatorProps {
 }
 
 /**
- * An AppRegistry construct to automatically create an application with the given name and description.
+ * An AppRegistry construct to automatically create applications with the given name and description
+ * in the desired regions.
  *
  * The application name must be unique at the account level and it's immutable.
+ * Only one application per region should be specified because a stack can only be associated with
+ * one application in a region.
  * This construct will automatically associate all stacks in the given scope, however
  * in case of a `Pipeline` stack, stage underneath the pipeline will not automatically be associated and
  * needs to be associated separately.
  *
+ * If cross region stack is detected, then this construct will give a warning and skip that association.
+ * Only stacks and an application within the same region will be associated.
+ *
```

Comment:
This should throw an error

Reply:
If this throws an error, all associations will stop.

We need to show a warning because we want to continue with associating all stacks with their respective regional applications. In the current implementation of stack-associator using Aspects, we are visiting all stacks regardless of their environment, and if the region matches then we associate the stack with application.